channeldb/graph.go

```diff
@@ -2824,6 +2864,20 @@ func putChanEdgePolicy(edges *bolt.Bucket, edge *ChannelEdgePolicy, from, to []b
 	return edges.Put(edgeKey[:], b.Bytes()[:])
 }
 
+// putChanEdgePolicyUnknown marks the edge policy as unknown in the edges bucket.
+func putChanEdgePolicyUnknown(edges *bolt.Bucket, channelID uint64, from []byte) error {
```

Comment:
Not needed with the new lookup.

Reply:
Based on your comments, I think we are not completely on the same page yet with regards to the db structure.

To get it completely clear, consider the following situation:
- Node A and B have a channel open. Channel id is 1
- Only the policy of node A is known

On master, this is represented in `edgeBucket` as:
`A | 1 -> policy`

When the backwards routing algorithm arrives at node B, it will need to know it's incoming edges. It looks up B in `edgeBucket` and finds nothing, even though there is a channel and known policy to reach B.

In this PR, the db structure is changed to:
`A | 1 -> policy`
`B | 1 -> unknown`

Now, the channel id can be retrieved when looking up B and the backward path can be continued.

You suggest that migration and putting the unknown values is not necessary. But this would mean that there will actually be no change to the db structure at all and the backward lookup problem will not be solved?

Reply:
Definitely, you are right! My understanding is that with this change DB structure should be:

`A | 1 -> <any value>`
`B | 1 -> <any value>`

So yes, you are right, we need to put a value for `B | 1` in case the policy is not there from before :)

Reply:
And also when a new channel is announced without any policies yet, we need to create two entries with unknown values. 

Ok, and `<any value>` in case of unknown: currently it is `[]byte {0}`. I think that if we want, I can change it into `[]byte {}`, the empty slice. What would be better / does it matter?

Reply:
Using the empty slice sounds good to me :)

Reply:
Changed